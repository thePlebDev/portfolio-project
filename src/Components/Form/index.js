import React from 'react';
import styled from 'styled-components';


const Form = styled.form`
  width:75%;
  margin:0 auto;
  margin-bottom:10%;
  position:relative;
`


const FormComponent =({children,onSubmit})=>{
  return(
    <Form onSubmit={(e)=>onSubmit(e)} data-testid="container">
      {children}
    </Form>
  )
}

export default FormComponent
