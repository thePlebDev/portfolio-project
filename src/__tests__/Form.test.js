import React from 'react';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Form from '../Components/Form'



describe('testing the Form component',()=>{
  it('should test the snapshot',()=>{
    const tree = renderer.create(<Form/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should test everything is rendered properly',()=>{
    const wrapper = shallow(<Form/>)
    expect(wrapper.find('[data-testid="container"]').length).toBe(1)
  })
  it('should test the submit function',()=>{
    const mockSubmitFunc = jest.fn(()=>{})
    const wrapper = shallow(<Form onSubmit={mockSubmitFunc}/>)
    const form = wrapper.find('[data-testid="container"]')
    form.simulate('submit')
    expect(mockSubmitFunc.mock.calls.length).toBe(1)
  })
})
