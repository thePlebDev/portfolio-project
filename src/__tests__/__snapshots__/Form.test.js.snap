// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the Form component should test the snapshot 1`] = `
<form
  className="sc-bdnylx kFpyQD"
  data-testid="container"
  onSubmit={[Function]}
/>
`;
